qiskit_experiments/calibration/calibrations_manager.py

```diff
@@ -0,0 +1,742 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Class to store and manage the results of a calibration experiments."""
+
+from collections import namedtuple, defaultdict
+from datetime import datetime
+from typing import Any, Dict, Set, Tuple, Union, List, Optional
+import csv
+import dataclasses
+import regex as re
+
+from qiskit.pulse import Schedule, ScheduleBlock, DriveChannel, ControlChannel, MeasureChannel, Call
+from qiskit.pulse.channels import PulseChannel
+from qiskit.circuit import Parameter, ParameterExpression
+from qiskit_experiments.calibration.exceptions import CalibrationError
+from qiskit_experiments.calibration.parameter_value import ParameterValue
+
+ParameterKey = namedtuple("ParameterKey", ["schedule", "parameter", "qubits"])
+ScheduleKey = namedtuple("ScheduleKey", ["schedule", "qubits"])
+
+
+class CalibrationsManager:
+    r"""
+    A class to manage schedules with calibrated parameter values. Schedules are
+    intended to be fully parameterized, including the index of the channels.
+    This class:
+    - supports having different schedules share parameters
+    - allows default schedules for qubits that can be overridden for specific qubits.
+
+    Parametric channel naming convention.
```

Comment:
Should a lot of the content in this docstring move up to a module level docstring?

Reply:
Sounds good. See here: 62b42b6